### PR TITLE
Silver berry and rainbow berry

### LINKED ISSUE #5: Silver berries
goldens, but for each part rather than for the entire map

*Migrated from iamdadbod's wishlist.*

### PR DESCRIPTION
**Pull request by maddie480**
tl;dr:
- Silver berries behave just like golden berries, except they have a different sprite and collect sound. The death sound also changes compared to the gold berry death sound.
- Rainbow berries behave just like red berries, except they have a different sprite and collect sound. They also don't spawn if you didn't collect all silver berries in a certain level set.

Because those berries behave just like vanilla ones, I went for the "extend Strawberry and hook when necessary" strat. Most hooks are shared between silver and rainbow berry, and one of them (death sound) will be re-used for speed berry.

Sprites and visual details are subject to change, but the functionality isn't, so I'm opening the PR now!

### CODE REVIEW

**Comment by 0x0ade** [comment]
> Sprites and visual details are subject to change, but the functionality isn't, so I'm opening the PR now!

Should this PR be converted into a draft PR or is it ready for review + merge?

**Comment by maddie480** [comment]
Changes will happen to the rainbow berry (they will get new sprites and we will replace the "1000" points with something else), but it already works as is. Should we turn that PR into a draft until all that is done, or should we merge a first version and make another PR when sprites and such are ready?

**Comment by 0x0ade** [comment]
We can merge this PR once it's usable, and then polish / fix things up afterwards. I guess getting this into the hands of mappers is a higher priority than getting it "perfect" first.

**Review by 0x0ade — APPROVED** [review]

**Review by Cruor — APPROVED** [review]
Looks good to me :+1:
